frontend-ssr/components/search/Search.tsx

```diff
@@ -0,0 +1,58 @@
+import React from 'react';
+
+import styles from './Search.module.scss';
+
+interface State {
+  searchValue: string;
+}
+interface Props {
+  query: string;
+  key: string;
+  onSearch: (e : any) => Promise<void>;
+}
+
+class Search extends React.Component<Props, State> {
+  constructor(props) {
+    super(props);
+
+    this.state = {
+      searchValue: props.query,
+    };
+  }
+
+  onChange = (e) => {
+    e.preventDefault();
+
+    this.setState({
+      searchValue: e.target.value,
+    });
+  };
+
+  render() {
+    const { searchValue } = this.state;
+    const { onSearch } = this.props;
+
+    return (
+      <div>
+        <form
+          onSubmit={onSearch}
+          className={`${styles.form} d-flex align-items-center`}
+        >
+          <input
+            type="text"
+            name="query"
+            placeholder="search"
+            value={searchValue}
+            onChange={this.onChange}
+            required
```

Comment:
This should not be required. Otherwise we're not going to be able to reset the search.